### ISSUE TITLE
doc: suggest: restrore the "defined type" terminology to "named type".

### ISSUE THREAD

**Issue by go101** [golang/go#35091]
Since the introduction of type alias, most places of using the old terminology `named type` are changed to `defined type`, to avoid some supposed confusions. Now it looks the old terminology `named type` is intended to be used to represent defined types or type aliases, but this is not confirmed in any official Go document.

However, this change brings [some embarrassing situations](https://github.com/golang/example/tree/master/gotypes#named-types), and causes [some inconveniences](https://github.com/golang/go/issues/32496) in [explaining some concepts](https://github.com/golang/go/issues/22005).

IMHO, I think this change is not very necessary. We can re-clarify that if we say `a type alias T is a named type`, we actually means `the type represented by the alias T is a named type`. If `T` represents an unnamed type, we should never say `T` is a named type, even if the alias `T` itself has a name.

If most people think my opinion is reasonable, I think it is still not too late to restrore the `defined type` terminology to `named type` now, for the terminology `defined type` is still not used widely but the terminology `named type` is still used widely.

In short, the benefits of this suggestion:
* avoids the embarrassing situation mentioned above
* avoids causing some inconveniences in explanations.
* **keep many many old documents/tutorials/apis still legal**.

**Comment by go101** [comment]
@griesemer @robpike how do you think?

**Comment by griesemer** [comment]
I'm concerned that switching back is only going to increase confusion. I leave this open for now so others can chime in, but there's no urgent need to change this.

I agree that _defined_ type is not great, but _named_ type is not great either in the presence of alias types. It is difficult to argue that `T` as declared in `type T struct { f int }` doesn't have a name (clearly it does have the name `T`) and thus is not a _named type_ (if it has a name, why is it not a named type?).

**Comment by go101** [comment]
@griesemer Did you mistyped `type T = struct { f int }` as `type T struct { f int }` in your last comment?

I think we can avoid calling aliases as types and only say aliases represents types to rationalize the restore.

**Comment by ianlancetaylor** [comment]
Aliases are definitely types.  You can use an alias anywhere you can use any other type.  I think that to say that aliases are not types would just make these concepts more complicated.

**Comment by griesemer** [comment]
@go101 Yes, it should have been `type T = struct { f int }`. Corrected.

**Comment by go101** [comment]
@ianlancetaylor Comparing to Go 1.8, it is surely a little more complicated. It is hard to get a perfect solution. Personally, I think it is better than the `defined type` solution, because keeping old documents/tutorials/apis still valid is a big benefit.

**Comment by go101** [comment]
As Go 1.18 added back the `named type` terminology, I will this issue soon.

**Comment by go101** [comment]
I understand. After all, many problems mentioned in this issue has gone with the change.
